monkey_island/cc/services/config.py

```diff
@@ -860,27 +876,38 @@ def creds_add_ntlm_hash(ntlm_hash):
         ConfigService.add_item_to_config_set('internal.exploits.exploit_ntlm_hash_list', ntlm_hash)
 
     @staticmethod
-    def update_config(config_json):
+    def update_config(config_json, should_encrypt):
+        if should_encrypt:
+            ConfigService.encrypt_config(config_json)
         mongo.db.config.update({'name': 'newconfig'}, {"$set": config_json}, upsert=True)
 
     @staticmethod
-    def get_default_config():
-        defaultValidatingDraft4Validator = ConfigService._extend_config_with_default(Draft4Validator)
-        config = {}
-        defaultValidatingDraft4Validator(SCHEMA).validate(config)
+    def init_default_config():
+        if ConfigService.default_config is None:
+            defaultValidatingDraft4Validator = ConfigService._extend_config_with_default(Draft4Validator)
+            config = {}
+            defaultValidatingDraft4Validator(SCHEMA).validate(config)
+            ConfigService.default_config = config
+
+    @staticmethod
+    def get_default_config(should_decrypt=True):
```

Comment:
Fun with naming
`should_decrypt` leads to `encrypt_config`, this does not make sense.

Reply:
Fixed